### ISSUE TITLE
Add rule: prefer-readonly-parameters

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#7]
## Problem

LLMs often mutate function parameters instead of creating new values.

## Rule

Enforce readonly on function parameters or flag parameter reassignment.

### ❌ Bad
```ts
function process(items: Item[]) {
  items.push(newItem);
  return items;
}
```

### ✅ Good
```ts
function process(items: readonly Item[]) {
  return [...items, newItem];
}
```

## Why it matters

- Prevents side effects
- Makes data flow explicit
- Easier to reason about purity
